App/Services/EventHandlers/UploadEventHandler.ts

```diff
@@ -26,16 +26,16 @@ export default class UploadEventHandler {
     } else if (responseCode === 401) {
       this.store.dispatch(ProcessingImagesActions.expiredTokenError(e.id))
     } else {
-      this.store.dispatch(ProcessingImagesActions.error(e.id, `Response code: ${responseCode}`))
+      this.store.dispatch(ProcessingImagesActions.uploadError(e.id, `Response code: ${responseCode}`))
```

Comment:
Dispatch upload-specific error actions for upload errors now.

Reply:
so does it also kills all other in-flight from that job if one errors?

Reply:
Ah good question... I guess in that case, i need to retry any uploads that have errored out. Let me look into that now.

Reply:
Ok, spent some time figuring this out. Current solution is... If one of the uploads from a share fails, it allows the other uploads to continue, but marks the image share as in an 'error' state and can be retried. The retry logic examines the state of each upload involved and kicks off new uploads for any upload that hasn't started or is in an error state.

Reply:
that's great